openbook_auth/models.py

```diff
@@ -3328,6 +3356,22 @@ def _check_comments_enabled_for_post_with_id(self, post_id):
                     _('Comments are disabled for this post')
                 )
 
+    def _check_can_translate_post_with_id(self, post_id):
+        Post = get_post_model()
+        post = Post.objects.get(id=post_id)
+        if post.text is None:
+            raise ValidationError(
+                _('Post has no text to be translated')
+            )
+        if post.language is None:
+            raise ValidationError(
+                _('Post has no assigned language to be able to translate')
+            )
+        if self.translation_language is None:
```

Comment:
Well yes it can happen not in the happy flow but for eg.
1. If a user who has not updated the app ( n therefore has no `translation_language` set, can still call the API with their auth token for eg via postman. IMO its expected not to throw 500s for such errors.
2. A language might be decommissioned, for eg. we lose a volunteer for faroese and there is no replacement to translate new strings so we decide to decommission it, we could remove it from supported locales or might be for some other reason we even decide to delete the language obj from the fixture, which will set these values to null. (on_delete is set_null)

Reply:
Cool, thanks!